pkg/agent/datamodel/sig_config.go

```diff
@@ -171,31 +96,74 @@ func (d Distro) IsContainerdDistro() bool {
 }
 
 func (d Distro) IsGPUDistro() bool {
-	for _, distro := range AvailableGPUDistros {
+	availableGPUDistros := []Distro{
+		AKSUbuntuGPU1804,
+		AKSUbuntuGPU1804Gen2,
+		AKSUbuntuGPUContainerd1804,
+		AKSUbuntuGPUContainerd1804Gen2,
+		AKSUbuntuFipsGPUContainerd1804,
+		AKSUbuntuFipsGPUContainerd1804Gen2,
+	}
+
+	for _, distro := range availableGPUDistros {
 		if d == distro {
 			return true
 		}
 	}
 	return false
 }
 func (d Distro) IsGen2Distro() bool {
-	for _, distro := range AvailableGen2Distros {
+	availableGen2Distros := []Distro{
+		AKSUbuntu1804Gen2,
+		AKSUbuntuGPU1804Gen2,
+		AKSUbuntuContainerd1804Gen2,
+		AKSUbuntuGPUContainerd1804Gen2,
+		AKSUbuntuFipsContainerd1804Gen2,
+		AKSUbuntuFipsContainerd2004Gen2,
+		AKSUbuntuFipsGPUContainerd1804Gen2,
+		AKSUbuntuEdgeZoneContainerd1804Gen2,
+		AKSUbuntuArm64Containerd1804Gen2,
+		AKSUbuntuArm64Containerd2204Gen2,
+		AKSUbuntuContainerd2204Gen2,
+		AKSUbuntuContainerd2004CVMGen2,
+		AKSUbuntuContainerd2204TLGen2,
+	}
+
+	for _, distro := range availableGen2Distros {
 		if d == distro {
 			return true
 		}
 	}
 	return false
 }
 func (d Distro) IsCBLMarinerDistro() bool {
-	for _, distro := range AvailableCBLMarinerDistros {
+	availableCBLMarinerDistros := []Distro{
+		AKSCBLMarinerV1,
+		AKSCBLMarinerV2,
+		AKSCBLMarinerV2Gen2,
+		AKSCBLMarinerV2Gen2Kata,
+		AKSCBLMarinerV2Arm64Gen2,
+		AKSCBLMarinerV2Gen2TL,
+		AKSCBLMarinerV2KataGen2TL,
+	}
+
+	for _, distro := range availableCBLMarinerDistros {
 		if d == distro {
 			return true
 		}
 	}
 	return false
 }
 func (d Distro) IsWindowsSIGDistro() bool {
-	for _, distro := range AvailableWindowsSIGDistros {
+	availableWindowsSIGDistros := []Distro{
```

Comment:
if you define these outside the func as un-exported, it still complains?

Reply:
Can you expand on what you mean by an unexported variable? From what I can tell, you would define the variable using a lower case letter but then need a function to set the value? - https://www.ardanlabs.com/blog/2014/03/exportedunexported-identifiers-in-go.html


```
type availableCBLMarinerDistros []Distro
func getAvailableCBLMarinerDistros( []Distro distro)  []Distro {
    return availableCBLMarinerDistros(distro) 
}

```

Reply:
> if you define these outside the func as un-exported, it still complains?

yeah it does unfortunately, I think I'll stick with the original format and add some nolint directives